src/body_est/src/body_est/realsense.py

```diff
@@ -215,21 +171,52 @@ def pose_estimate_body(self, landmarks_list):
         # get landmarks coords
         for lm in self.lm_body:
             # transform landmark to 3D pose and transform coords to base link
-            landmarks[lm.value]["point"] = self.transform_point_baselink(
+            landmarks[lm.value]["point"] = self.transform_point_cameralink(
                 self.landmark_to_3d(landmarks[lm.value]["point"])
             )
-            # visualize points in Rviz
-            self.point_viz_pub.publish(landmarks[lm.value]["point"])
-
-        # rospy.loginfo(f"Transformed {landmarks}")
 
-    def compute_unit_normal(self, landmarks, align_vertical=False):
+        # process landmarks a bit
+        self.point_viz_pub.publish(landmarks[Landmarks.RIGHT_SHOULDER.value]["point"])
+        self.point_viz_pub.publish(landmarks[Landmarks.LEFT_SHOULDER.value]["point"])
+        self.point_viz_pub.publish(landmarks[Landmarks.RIGHT_HIP.value]["point"])
+        self.point_viz_pub.publish(landmarks[Landmarks.LEFT_HIP.value]["point"])
+
+        # returns a dictionary of the landmarks, their info, and their 3D coordinate
+        unit_nrml_x = self.compute_unit_normal(landmarks)
+        unit_nrml_y = np.array([-unit_nrml_x[1], unit_nrml_x[0], 0])
+        unit_nrml_z = np.cross(unit_nrml_x, unit_nrml_y)
+
+        rospy.loginfo(f"x {unit_nrml_x} y {unit_nrml_y} z {unit_nrml_z}")
+        rospy.loginfo(f"xy_dot {np.dot(unit_nrml_x,unit_nrml_y)}")
+        rospy.loginfo(f"xz_dot {np.dot(unit_nrml_x,unit_nrml_z)}")
+        rospy.loginfo(f"yz_dot {np.dot(unit_nrml_y,unit_nrml_z)}")
+
+        alpha = self.compute_euler_ang(unit_nrml_x, self.x_unit)
```

Comment:
Are you sure about this conversion from axis angles to Euler angles and quaternions is correct? There is an implied order with the equations here, so it would be helpful to make sure that the quaternion that results from this routine is the expected one.

Reply:
I'll look into it